http2_general/test_h2_block_action.py

```diff
@@ -142,19 +242,198 @@ def test_block_action_attack_reply_not_on_req_rcv_event(self):
+    tempesta_tmpl = """
+        listen 443 proto=h2;
+        srv_group default {
+            server ${server_ip}:8000;
+        }
+        vhost good {
+            proxy_pass default;
+        }
+        vhost frang {
+            frang_limits {
+                http_methods GET;
+                http_resp_code_block 200 1 10;
+            }
+            proxy_pass default;
+        }
+        tls_certificate ${tempesta_workdir}/tempesta.crt;
+        tls_certificate_key ${tempesta_workdir}/tempesta.key;
+        tls_match_any_server_name;
+
+        block_action attack %s;
+        block_action error %s;
+        response_body 4* %s;
+
+        http_chain {
+            host == "bad.com"   -> block;
+            host == "good.com"  -> good;
+            host == "frang.com" -> frang;
+        }
+    """
+
+    ERROR_RESPONSE_BODY = "a" * 1000
+    INITIAL_WINDOW_SIZE = 65535
+
+    def __generate_custom_error_page(self, data):
+        workdir = tf_cfg.cfg.get("General", "workdir")
+        cpage_gen = CustomErrorPageGenerator(data=data, f_path=f"{workdir}/4xx.html")
+        path = cpage_gen.get_file_path()
+        remote.tempesta.copy_file(path, data)
+
+        return path
+
+    def setUp(self):
+        path = self.__generate_custom_error_page(self.ERROR_RESPONSE_BODY)
+        self.tempesta = {
+            "config": self.tempesta_tmpl % ("reply", "reply", path),
+        }
+        H2Base.setUp(self)
+
+
+class BlockActionH2ReplyWithCustomErrorPageSmallWindow(BlockActionH2ReplyWithCustomErrorPage):
+    ERROR_RESPONSE_BODY = "a" * 1000
+    INITIAL_WINDOW_SIZE = 100
+
+    def test_block_action_error_reply_with_conn_close_multiple_requests(self):
+        client = self.get_client("deproxy")
+
+        sniffer = self.setup_sniffer()
+        self.start_services_and_initiate_conn(client)
+        self.save_must_fin_socks([client])
+        self.save_must_not_reset_socks([client])
+
+        good_req = [
+            HeaderTuple(":authority", "good.com"),
+            HeaderTuple(":path", "/"),
+            HeaderTuple(":scheme", "https"),
+            HeaderTuple(":method", "GET"),
+        ]
+        curr_responses = len(client.responses)
+
+        client.make_requests(
+            requests=[
+                [
+                    HeaderTuple(":authority", "good.com"),
+                    HeaderTuple(":path", "/"),
+                    HeaderTuple(":scheme", "https"),
+                    HeaderTuple(":method", "GET"),
+                    HeaderTuple("Content-Type", "invalid"),
+                ],
+                good_req,
+                good_req,
+                good_req,
+            ]
+        )
+        client.wait_for_response(3)
+
+        self.assertEqual(curr_responses + 1, len(client.responses))
+        self.assertEqual(client._last_response.status, "400")
+        self.assertEqual(client._last_response.body, self.ERROR_RESPONSE_BODY)
+        self.assertTrue(client.wait_for_connection_close())
```

Comment:
This should speed up test
```suggestion
        self.assertTrue(client.wait_for_connection_close())

        self.assertEqual(curr_responses + 1, len(client.responses))
        self.assertEqual(client.last_response.status, "400")
        self.assertEqual(client.last_response.body, self.ERROR_RESPONSE_BODY)
```

or make `client.valid_req_num = 1` before `client.wait_for_response(3)`